packages/modules/src/modules/epics/NewsletterSignup.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { css } from '@emotion/react';
+import { space } from '@guardian/src-foundations';
+
+const containerStyles = css`
+    margin: ${space[6]}px ${space[2]}px ${space[1]}px 0;
+    width: 100%;
+`;
+
+const NewsletterSignup = ({ url }: { url: string }): JSX.Element => {
+    const [iframeHeight, setIframeHeight] = useState(60);
+    const iframeRef = useRef<HTMLIFrameElement>(null);
+
+    useEffect(() => {
+        // Handle iframe resize events. Based on https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/browser/newsletterEmbedIframe/init.ts
+        window.addEventListener('message', event => {
+            try {
+                // Check if this is the newsletter iframe
+                const contentWindow = iframeRef?.current?.contentWindow;
+                if (contentWindow && event.source && contentWindow === event.source) {
+                    const message = JSON.parse(event.data);
+
+                    if (message.type === 'set-height') {
+                        if (typeof message.value === 'number') {
+                            setIframeHeight(message.value);
+                        } else if (typeof message.value === 'string') {
+                            const value = parseInt(message.value, 10);
```

Comment:
I think passing a number to `parseInt` will happy return a number, so I wonder if we could lose/combine some of the conditions here to simplify things (i.e. call `parseInt` in either case):

```js
parseInt(50, 10); // => 50
```

Reply:
that sounds possible! This part was heavily based on the tried + tested [dotcom implementation](https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/browser/newsletterEmbedIframe/init.ts#L45), so I'm reluctant to deviate!